test/libsolidity/Imports.cpp

```diff
@@ -104,8 +104,7 @@ BOOST_AUTO_TEST_CASE(simple_alias)
 BOOST_AUTO_TEST_CASE(library_name_clash)
 {
 	CompilerStack c;
-	c.addSource("a", "library A {} pragma solidity >=0.0;");
-	c.addSource("b", "library A {} pragma solidity >=0.0;");
+	c.addSource("a", "library A {} library A {} pragma solidity >=0.0;");
```

Comment:
Right, I would also prefer to use fully-qualified names for the linker. In the future, libraries will be linked differently and you do not have the 36 character limit. Also, the format of using `__A____` is only a semi-backwards-compatible hack. You should really use `solc --libraries my/file/name.sol:A=0x123 my/contract.sol` for compiling, and there, there is obviously not such a small size limit.

If you want, you can merge this pull request and do the library handling in another one.